saged/predict_tissue.py

```diff
@@ -28,6 +30,70 @@
                      'Cerebellum', 'Cerebral_cortex', 'Salivary_Gland', 'Duodenum'
                      ]
 
+
+def objective(trial, train_list, supervised_config,
+              label_encoder, weighted_loss=False, device='cpu'):
+    losses = []
+
+    with open(supervised_config) as supervised_file:
+        supervised_config = yaml.safe_load(supervised_file)
+    supervised_model_type = supervised_config.pop('name')
+
+    # TODO there should be a better way to do this which allows future skl models
+    if supervised_model_type != 'LogisticRegression':
+        lr = trial.suggest_float('lr', 1e-6, 10, log=True)
+    l2_penalty = trial.suggest_float('l2_penalty', 1e-6, 10, log=True)
+
+    for i in range(len(train_list)):
+        inner_train_list = train_list[:i] + train_list[i+1:]
+        LabeledDatasetClass = type(labeled_data)
+
+        inner_train_data = LabeledDatasetClass.from_list(inner_train_list)
+        inner_val_data = train_list[i]
+        inner_train_data.set_label_encoder(label_encoder)
+        inner_val_data.set_label_encoder(label_encoder)
+
+        # Sklearn logistic regression doesn't allow manually specifying classes
+        # so we have to do this
+        if len(inner_train_data.get_classes()) < len(inner_val_data.get_classes()):
+            continue
+
+        input_size = len(inner_train_data.get_features())
+        output_size = len(label_encoder.classes_)
+
+        with open(args.supervised_config) as supervised_file:
+            supervised_config = yaml.safe_load(supervised_file)
+            supervised_config['input_size'] = input_size
+            supervised_config['output_size'] = output_size
+            supervised_config['log_progress'] = False
+            supervised_config['l2_penalty'] = l2_penalty
+            if supervised_model_type != 'LogisticRegression':
```

Comment:
Maybe you could define a list of models that don't have the `lr` parameter in one place (like at the top of the function), then just check against it both here and on line 42?

Just an idea - you only have this check in two places so it's not a huge deal, I'm fine with keeping it like this if you want.

Reply:
Good idea, I'll add it if I need to include more than one model type. As it is right now it's very clearly bad, where if I was comparing to a const I might think that I had a good solution :)